### PR TITLE
feat: Definition to disable JSON auto parsing

### PR DESCRIPTION
**Pull request by Logan-seongjae**
issue : https://github.com/hiro5id/console-log-json/issues/41

- code changes
  - Added definition option to disable JSON auto parsing.
    - `CONSOLE_LOG_JSON_DISABLE_AUTO_PARSE`
  - lint test code

### CODE REVIEW

**Comment by hiro5id** [comment]
thank you so much for the PR... i'll take a look right away

File: src/logger.ts [hunk 1]
```diff
@@ -172,14 +172,25 @@ export function FormatErrorObject(object: any) {
...
+    const { CONSOLE_LOG_JSON_DISABLE_AUTO_PARSE } = process.env;
     try {
-      parsedObject = JSON.parse(returnData.message);
+      // if defined CONSOLE_LOG_JSON_DISABLE_AUTO_PARSE=TRUE, disable auto parsing.
+      if (!CONSOLE_LOG_JSON_DISABLE_AUTO_PARSE) {
+        parsedObject = JSON.parse(returnData.message);
+      } else {
+        parsedObject = JSON.parse(returnData.message); // trim & remove new lines
+        parsedObject = JSON.stringify(parsedObject);
+      }
```
Comment (hiro5id) [inline src/logger.ts:183]:
```suggestion
      if (CONSOLE_LOG_JSON_DISABLE_AUTO_PARSE) {
        parsedObject = JSON.parse(returnData.message); // trim & remove new lines
        parsedObject = JSON.stringify(parsedObject);
      } else {
        parsedObject = JSON.parse(returnData.message);
      }
```

Please flip if-else statements to simplify.

File: src/logger.ts [hunk 2]
```diff
@@ -172,14 +172,25 @@ export function FormatErrorObject(object: any) {
...
     }
     if (parsedObject != null) {
-      returnData.message = '<auto-parsed-json-string-see-@autoParsedJson-property>';
-      returnData['@autoParsedJson'] = parsedObject;
+      if (!CONSOLE_LOG_JSON_DISABLE_AUTO_PARSE) {
+        returnData.message = '<auto-parsed-json-string-see-@autoParsedJson-property>';
+        returnData['@autoParsedJson'] = parsedObject;
+      } else {
+        returnData.message = parsedObject;
+      }
```
Comment (hiro5id) [inline src/logger.ts:193]:
```suggestion
      if (CONSOLE_LOG_JSON_DISABLE_AUTO_PARSE) {
        returnData.message = parsedObject;
      } else {
        returnData.message = '<auto-parsed-json-string-see-@autoParsedJson-property>';
        returnData['@autoParsedJson'] = parsedObject;
      }
```

Please flip if-else statements to simplify.

**Review by hiro5id — CHANGES_REQUESTED** [review]
Thank you so much for the contribution. I have suggested a very minor change. I will merge as soon as it is ready. Thanks again!

**Review by hiro5id — APPROVED** [review]

File: src/logger.ts (hunk 1 above)
Comment (Logan-seongjae) [reply src/logger.ts]: so good thanks for your review.

**Comment by hiro5id** [comment]
@Logan-seongjae  These are deployed in v2.10.0 and up on NPM by the way..  I appreciate taking your time to contribute.

@all-contributors please add @Logan-seongjae for JSON Parsing enhancements

**Comment by hiro5id** [comment]
https://github.com/all-contributors please add @Logan-seongjae for code

**Comment by hiro5id** [comment]
@all-contributors please add @Logan-seongjae for code

**Pull request merged**